packages/cli/internal/pkg/cli/logs_adapter.go

```diff
@@ -35,6 +38,22 @@ func (o *logsAdapterOpts) Validate() error {
 		return err
 	}
 
+	// It's convenient for the tests to be able to not actually have the ctxManager set.
```

Comment:
I think this comment doesn't add much information, and can be removed in order to keep code clean.

Reply:
Do we need to explain somewhere else why we need to support `ctxManager` being `nil` when it isn't ever in normal operation? Or do we just let the tests constrain future developers who might think it was safe to remove the check?

Reply:
It is a very good point.
I think that the fact, that we have to "explain" something is a red flag. It meas 1) code is not clear and self explanatory 2) we are doing something wrong.

There are 2 questions here.
1) is `nil` a valid state of a context manager? It is not. It means that ignoring that fact is not a correct behavior of the "validate" method, because it can produce unexpected and hard to troubleshoot problems in production
2) Should business logic be tailored to unit test? No, it should not. It's effectively makes unit testing less efficient, and add complexity to our main logic. 

My recommendation here is to remove this check, and make sure we properly populate ctxManager in tests. In this case we:
1) don't mix test logic with business logic
2) do not introduce any implicit dependency that has to be explained in a comment or documentation
3) make sure we are testing actual code flow in our unit tests.

Reply:
OK, I've set up the tests for adapter logs to all have a stub context manager, that will return contexts if requested but that won't insist on being called. I think that's the contract we want fulfilled here.

Then in `Validate()` I'm making sure that the context manager is set.

So the tests now pass and should keep passing even if the context manager checks move relative to the other validation checks, and tests start/stop getting the context.